dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java

```diff
@@ -0,0 +1,181 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.common.metrics.collector;
+
+import org.apache.dubbo.common.metrics.event.BaseMetricsEvent;
+import org.apache.dubbo.common.metrics.event.NewRTEvent;
+import org.apache.dubbo.common.metrics.event.NewRequestEvent;
+import org.apache.dubbo.common.metrics.listener.MetricsListener;
+import org.apache.dubbo.common.metrics.model.MethodMetric;
+import org.apache.dubbo.common.metrics.model.sample.GaugeMetricSample;
+import org.apache.dubbo.common.metrics.model.sample.MetricSample;
+import org.apache.dubbo.rpc.model.ApplicationModel;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.apache.dubbo.common.metrics.model.MetricsCategory.REQUESTS;
+import static org.apache.dubbo.common.metrics.model.MetricsCategory.RT;
+
+/**
+ * Default implementation of {@link MetricsCollector}
+ */
+public class DefaultMetricsCollector implements MetricsCollector {
+
+    private Boolean collectEnabled = false;
+    private final List<MetricsListener> listeners = new ArrayList<>();
+    private final ApplicationModel applicationModel;
+    private final String applicationName;
+
+    private final Map<MethodMetric, AtomicLong> totalRequests = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> succeedRequests = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> failedRequests = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> processingRequests = new HashMap<>();
+
+    private final Map<MethodMetric, AtomicLong> lastRT = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> minRT = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> maxRT = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> avgRT = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> totalRT = new HashMap<>();
+    private final Map<MethodMetric, AtomicLong> rtCount = new HashMap<>();
+
+    public DefaultMetricsCollector(ApplicationModel applicationModel) {
+        this.applicationModel = applicationModel;
+        this.applicationName = applicationModel.tryGetApplicationName();
```

Comment:
The `applicationName` is required,  should not use `applicationModel.tryGetApplicationName()`